lib/src/rules/unnecessary_final.dart

```diff
@@ -85,9 +85,9 @@ class _Visitor extends SimpleAstVisitor<void> {
   }
 
   @override
-  void visitVariableDeclaration(VariableDeclaration node) {
-    if (node.isFinal) {
-      rule.reportLint(node.name);
+  void visitVariableDeclarationStatement(VariableDeclarationStatement node) {
+    if (node.variables.isFinal) {
+      rule.reportLint(node.variables);
```

Comment:
Unlike the previous implementation, this has the potential of highlighting a significantly long range of characters. Consider changing this (and other places the lint is reported for the sake of consistency) to only highlight the keyword `final`.